### ISSUE TITLE
Doesn't work for more than 10 speakers.

### ISSUE THREAD

**Issue by PARULSHARM** [Adirockzz95/Piwho#6]
If i use more than 10 speakers for given code.
```
from piwho import recognition
recog = recognition.SpeakerRecognizer()
name = []
name = recog.identify_speaker('audio.wav')
print(name[0])
print(name[1])


```
Then rather than getting speaker's name, I get this output.
`[options]`
Please tell me what's the reason.

Thanks.

**Comment by Adirockzz95** [comment]
add ``recog.debug = True`` before the ``name = recog.identify_speaker('audio.wav')``. This will print debug messages.

**Comment by PARULSHARM** [comment]
After putting recog.debug=True, I get this output.

`Error opening speaker DB: "speakers.txt": speakers.txt (No such file or directory).
Traceback (most recent call last):
Error opening speaker DB: "speakers.txt": speakers.txt (No such file or directory).
  File "/home/parul/PycharmProjects/back/piwho-bak/test.py", line 30, in <module>
    print(name[0])
	at SpeakersIdentD.connect(SpeakersIdentD.java:179)
	at SpeakerIdentApp.main(SpeakerIdentApp.java:298)
TypeError: 'NoneType' object has no attribute '__getitem__'
SpeakersIdentD.close() - not connected
	at SpeakersIdentD.close(SpeakersIdentD.java:282)
	at SpeakerIdentApp.main(SpeakerIdentApp.java:601)

`

**Comment by Adirockzz95** [comment]
Do you have speakers.txt present in your current directory?

**Comment by PARULSHARM** [comment]
Yes ! I have.
After adding 

`recog.set_feature_option('-wav -silence -noise -lpc -che -deug')
`
I get this output.

```
For input string: ":"
java.lang.NumerFormatException: For input string: ":"
	at java.lang.NumerFormatException.forInputString(NumerFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at SpeakersIdentD.query(SpeakersIdentD.java:213)
	at SpeakerIdentApp.main(SpeakerIdentApp.java:299)
Usage:
  java SpeakerIdentApp --train <samples-dir> [options]        -- train mode
                       --single-train <sample> [options]      -- add a single sample to the training set
                       --ident <sample> [options]             -- identification mode
                       --atch-ident <samples-dir> [options]  -- atch identification mode
                       --gui                                  -- use GUI as a user interface
                       --stats=[per-config|per-speaker|oth]  -- display stats (default is per-config)
                       --est-score                           -- display est classification result
                       --reset                                -- reset stats
                       --version                              -- display version info
                       --help | -h                            -- display this help and exit

Options (one or more of the following):

Loaders:

  -wav          - assume WAVE files loading (default)
  -text         - assume loading of text samples

Preprocessing:

  -silence      - remove silence (can e comined with any of the elow)
  -noise        - remove noise (can e comined with any of the elow)
  -raw          - no preprocessing
  -norm         - use just normalization, no filtering
  -low          - use low-pass FFT filter
  -high         - use high-pass FFT filter
  -oost        - use high-frequency-oost FFT preprocessor
  -and         - use and-pass FFT filter
  -andstop     - use and-stop FFT filter
  -endp         - use endpointing
  -lowcfe       - use low-pass CFE filter
  -highcfe      - use high-pass CFE filter
  -andcfe      - use and-pass CFE filter
  -andstopcfe  - use and-stop CFE filter

Feature Extraction:

  -lpc          - use LPC
  -fft          - use FFT
  -minmax       - use Min/Max Amplitudes
  -randfe       - use random feature extraction
  -aggr         - use aggregated FFT+LPC feature extraction
  -f0           - use F0 (pitch, or fundamental frequency; NOT IMPLEMENTED)
  -segm         - use Segmentation (NOT IMPLEMENTED)
  -cepstral     - use Cepstral analysis (NOT IMPLEMENTED)

Classification:

  -nn           - use Neural Network
  -che         - use Cheyshev Distance
  -eucl         - use Euclidean Distance
  -mink         - use Minkowski Distance
  -diff         - use Diff-Distance
  -zipf         - use Zipfs Law-ased classifier
  -randcl       - use random classification
  -markov       - use Hidden Markov Models (NOT IMPLEMENTED)
  -hamming      - use Hamming Distance
  -cos          - use Cosine Similarity Measure

Misc:

  -deug        - include verose deug output
  -spectrogram  - dump spectrogram image after feature extraction
  -graph        - dump wave graph efore preprocessing and after feature extraction
  <integer>     - expected speaker ID



None

```
What is the issue and everytime i increase number of speakers more than 10. It gives abover error.

**Comment by PARULSHARM** [comment]
I tried this. Still, I'm getting the same error.

**Comment by Adirockzz95** [comment]
Also, make sure you're using the same feature_option for training and recognition.

**Comment by Adirockzz95** [comment]
If possible post your code.

**Comment by PARULSHARM** [comment]
Train file

`from piwho import recognition
recog = recognition.SpeakerRecognizer()
# recog.set_feature_option('-wav -silence -noise -lpc -che -deug')
recog.set_feature_option('-silence -noise -lpc -cheb')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/84/121123/84-121123-0025.wav', '84')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/84/121123/84-121123-0024.wav', '84')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/84/121123/84-121123-0023.wav', '84')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/84/121123/84-121123-0022.wav', '84')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/84/121123/84-121123-0021.wav', '84')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/84/121123/84-121123-0024.wav', '84')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0019.wav', '174')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0018.wav', '174')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0017.wav', '174')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0016.wav', '174')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0015.wav', '174')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0014.wav', '174')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/251/118436/251-118436-0000.wav', '251')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/251/118436/251-118436-0002.wav', '251')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/422/122949/422-122949-0000.wav', '422')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/422/122949/422-122949-0001.wav', '422')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/652/129742/652-129742-0000.wav', '652')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/652/129742/652-129742-0001.wav', '652')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/777/126732/777-126732-0000.wav', '777')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/777/126732/777-126732-0001.wav', '777')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1272/128104/1272-128104-0000.wav', '1272')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1272/128104/1272-128104-0001.wav', '1272')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1462/170138/1462-170138-0000.wav', '1462')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1462/170138/1462-170138-0001.wav', '1462')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1673/143396/1673-143396-0000.wav', '1673')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1673/143396/1673-143396-0001.wav', '1673')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1919/142785/1919-142785-0000.wav', '1919')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1919/142785/1919-142785-0001.wav', '1919')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1988/24833/1988-24833-0000.wav', '1988')
recog.train_new_data('/home/parul/Documents/Piwho/Piwho/dev-clean/1988/24833/1988-24833-0001.wav', '1988')


print(recog.get_speakers())
`

test file

`from piwho import recognition
recog = recognition.SpeakerRecognizer()
name = []
recog.debug = True
recog.set_feature_option('-silence -noise -lpc -cheb')
name = recog.identify_speaker('/home/parul/Documents/Piwho/Piwho/dev-clean/174/50561/174-50561-0009.wav')
# dictn = recog.get_speaker_scores()
# print dictn
print(name)`

**Comment by PARULSHARM** [comment]
This code works fine until 10 speakers but gives the following error when I give 11 speakers.
`For input string: ":"
java.lang.NumerFormatException: For input string: ":"
	at java.lang.NumerFormatException.forInputString(NumerFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at SpeakersIdentD.query(SpeakersIdentD.java:213)
	at SpeakerIdentApp.main(SpeakerIdentApp.java:299)
Usage:
  java SpeakerIdentApp --train <samples-dir> [options]        -- train mode
                       --single-train <sample> [options]      -- add a single sample to the training set
                       --ident <sample> [options]             -- identification mode
                       --atch-ident <samples-dir> [options]  -- atch identification mode
                       --gui                                  -- use GUI as a user interface
                       --stats=[per-config|per-speaker|oth]  -- display stats (default is per-config)
                       --est-score                           -- display est classification result
                       --reset                                -- reset stats
                       --version                              -- display version info
                       --help | -h                            -- display this help and exit

Options (one or more of the following):

Loaders:

  -wav          - assume WAVE files loading (default)
  -text         - assume loading of text samples

Preprocessing:

  -silence      - remove silence (can e comined with any of the elow)
  -noise        - remove noise (can e comined with any of the elow)
  -raw          - no preprocessing
  -norm         - use just normalization, no filtering
  -low          - use low-pass FFT filter
  -high         - use high-pass FFT filter
  -oost        - use high-frequency-oost FFT preprocessor
  -and         - use and-pass FFT filter
  -andstop     - use and-stop FFT filter
  -endp         - use endpointing
  -lowcfe       - use low-pass CFE filter
  -highcfe      - use high-pass CFE filter
  -andcfe      - use and-pass CFE filter
  -andstopcfe  - use and-stop CFE filter

Feature Extraction:

  -lpc          - use LPC
  -fft          - use FFT
  -minmax       - use Min/Max Amplitudes
  -randfe       - use random feature extraction
  -aggr         - use aggregated FFT+LPC feature extraction
  -f0           - use F0 (pitch, or fundamental frequency; NOT IMPLEMENTED)
  -segm         - use Segmentation (NOT IMPLEMENTED)
  -cepstral     - use Cepstral analysis (NOT IMPLEMENTED)

Classification:

  -nn           - use Neural Network
  -che         - use Cheyshev Distance
  -eucl         - use Euclidean Distance
  -mink         - use Minkowski Distance
  -diff         - use Diff-Distance
  -zipf         - use Zipfs Law-ased classifier
  -randcl       - use random classification
  -markov       - use Hidden Markov Models (NOT IMPLEMENTED)
  -hamming      - use Hamming Distance
  -cos          - use Cosine Similarity Measure

Misc:

  -deug        - include verose deug output
  -spectrogram  - dump spectrogram image after feature extraction
  -graph        - dump wave graph efore preprocessing and after feature extraction
  <integer>     - expected speaker ID



None
`

**Comment by Adirockzz95** [comment]
It seems that MARF's command line parser is throwing an error after 10 speakers. I'll look into it.

**Comment by PARULSHARM** [comment]
Okay ! Thanks

**Comment by Adirockzz95** [comment]
This issue has been resolved in v1.3.2.
